### ISSUE TITLE
Properties Editor - Hidden Tools tab issues?

### ISSUE THREAD

**Issue by Draise14** [Bforartists/Bforartists#1411]
# Issue
![firefox_2020-03-11_14-24-12](https://user-images.githubusercontent.com/35357409/76455748-05202300-63a4-11ea-8f9c-79e7856c363e.png)

There has been a comment about the Tools tab/panel now being missing in the Properties Editor (which we did on purpose as it is redundant), mentioned in the FB groups. It shows up on first use of the software, but then disappears when changing tabs. I can't reproduce it though... Maybe it's a setting migrated from BFA2 - A 0.6.0 (Blender 2.80)?  

![bforartists_2020-03-11_14-25-58](https://user-images.githubusercontent.com/35357409/76455963-4b758200-63a4-11ea-95cb-53fe1e17410e.png)

I have also noticed that some viewport overriding active tools makes the sidebar panel usability freeze. The Scatter Tool addon included with latest releases has this issue, where you can't change any value in the active tool because the widget overlay overrides any sidebar interactivity. The tool settings, in Blender, can be manipulated in the Properties Editor only live before you apply the scatter. This brings up issues with compatibility with other addons, will there also be other addons with viewport override modes and with editable properties? 

I know we discussed how to minimize these tools locations (twice in the 3D view and 1 in the Properties Editor,) but it has made some small issue... 

![firefox_2020-03-11_15-02-56](https://user-images.githubusercontent.com/35357409/76458993-6bf40b00-63a9-11ea-946a-ae059a684892.png)

# Questions
1. How do you get addons that override the viewport usability that go in some kind of "draw" mode where you can't access the tool properties in the Tool Shelf? 
2. Is this properties Tools tab worth developing to be optionally shown with a UI toggle somewhere? Is that even possible? 
3. Maybe having 2 tool settings (toolbar header and toolshelf) in the 3D view is not logical, and only having 1 (toolbar header) and then all tool settings in the properties tab might make more sense? 
4. The Tools tab of the Properties shelfof the 3D View , can that even be hidden technically (incase you'd want to toggle one or the other)? If so, maybe a toggle could be made to either show the active tool panels in the Toolshelf or Properties Editor Tools tab only. 

## Tasks
- Evaluate if hiding the toolshelf active tool panels/tab in the 3D View is better than hiding the tool tab in the Properties Editor
- Check if and when or how the tab still shows sometimes to make sure it doesn't.
- Add toggle to UI to optionally show one or the other active tools panels (3D View Properties Shelf only or Properties Editor only). _(oooonnnnly if it's possible or if it's hidden code and easy to implement)_

**Comment by ReinerBforartists** [comment]
Thanks for the report. 

Huh. I am not aware that you can reveal this tab again. The menu item is disabled. It should not even show with old settings or an addon. Let me investigate here what could be the cause. Maybe you can convince the user to show us a way to reproduce how he is able to resurrect the tab button.

For adding a toggle, hm. It's C code, I hate it ^^

I can understand that letting go here is a bit tough when you got used to use this area. But i don't want to go back really. This tab is redundant. We can't have the same content in all editors.

> I have also noticed that some viewport overriding active tools makes the sidebar panel usability freeze. The Scatter Tool addon included with latest releases has this issue, where you can't change any value in the active tool because the widget overlay overrides any sidebar interactivity. The tool settings, in Blender, can be manipulated in the Properties Editor only live before you apply the scatter. This brings up issues with compatibility with other addons, will there also be other addons with viewport override modes and with editable properties?

Could you please make an extra task for this? Sounds like something else, and has nothing to do with this issue here.

**Comment by ReinerBforartists** [comment]
> Evaluate if hiding the toolshelf active tool panels/tab in the 3D View is better than hiding the tool tab in the Properties Editor. 
This is not a solveable problem. The design decision was to remove the tools button in the properties editor.

>Check if and when or how the tab still shows sometimes to make sure it doesn't.
Checked, not to reproduce. The ghost tab button is not to reproduce. I have run through all possible scenarios now. The button is deactivated. It should simply not show.

>Add toggle to UI to optionally show one or the other active tools panels (3D View Properties Shelf only or Properties Editor only). (oooonnnnly if it's possible or if it's hidden code and easy to implement)

I can't do this. We won't reintroduce removed redundancies. Even when some got used to them meanwhile. The development goal is to remove such redundancies to have a cleaner user interface.

Bottom line: Still waiting for more data here. Maybe somebody finds a way to reproduce it that i have overlooked.

**Comment by Draise14** [comment]
I'll make the other report on the Scatter Addon not being able to tune from the properties shelf. 

Yeah, I kinda like not needing to think about the other panel. But nonetheless, I've asked for more info from the user. No answer yet.

**Comment by ReinerBforartists** [comment]
Any news about the issue that you wanted to create? :)

Okay. We have a task. I stumbled across tools settings in the factory defaults in one of the layouts. I make them new now anyways because of a merge bug. So this content will vanish.

![toolsettings](https://user-images.githubusercontent.com/13919702/76706881-15f5cf00-66eb-11ea-8e75-da17110b00ba.jpg)

**Comment by ReinerBforartists** [comment]
Related: Merge problem with new vertex paint feature for grease pencil #1412

**Comment by ReinerBforartists** [comment]
Recreated the layouts because of the bug. Means this issue should be fixed: Merge problem with new vertex paint feature for grease pencil #1412

Closing.
